File: Extensions/WrapLive WatchKit Extension/Categories/WKInterfaceController/WKInterfaceController+SimplifiedTextInput.m
//
//  WKInterfaceController+SimplifiedTextInput.m
//  wrapLive
//
//  Created by Sergey Maximenko on 6/16/15.
//  Copyright (c) 2015 Ravenpod. All rights reserved.
//

#import "WKInterfaceController+SimplifiedTextInput.h"

@implementation WKInterfaceController (SimplifiedTextInput)

- (void)presentTextInputControllerWithSuggestionsFromFileNamed:(NSString *)fileName completion:(void (^)(NSString *result))completion {
    NSString *path = [[NSBundle mainBundle] pathForResource:fileName ofType:@"plist"];
    NSArray *presets = [NSArray arrayWithContentsOfFile:path];
    [self presentTextInputControllerWithSuggestions:presets allowedInputMode:WKTextInputModeAllowEmoji completion:^(NSArray *results) {
        for (NSString *result in results) {
            if ([result isKindOfClass:[NSString class]]) {
                if (completion) completion(result);
                break;
            }
        }
    }];
}

@end

File: Extensions/WatchExt/Categories/WKInterfaceController+SimplifiedTextInput.swift
//
//  WKInterfaceController+SimplifiedTextInput.swift
//  meWrap
//
//  Created by Sergey Maximenko on 10/27/15.
//  Copyright © 2015 Ravenpod. All rights reserved.
//

import WatchKit

extension WKInterfaceController {
    func presentTextSuggestionsFromPlistNamed(name: String, completionHandler: (String -> Void)) {
        guard let path = NSBundle.mainBundle().pathForResource(name, ofType: "plist") else {
            return
        }
        let presets = NSArray(contentsOfFile: path) as? [String]
        presentTextInputControllerWithSuggestions(presets, allowedInputMode: .AllowEmoji) { (results) -> Void in
            guard let results = results else {
                return
            }
            for result in results {
                if let result = result as? String {
                    completionHandler(result)
                    break
                }
            }
        }
    }
}